src/main/java/com/minecraftabnormals/abnormals_core/core/util/BiomeUtil.java

```diff
@@ -102,6 +152,71 @@ public static synchronized void markEndBiomeCustomMusic(ResourceLocation biomeNa
 	public static boolean shouldPlayCustomEndMusic(ResourceLocation biomeName) {
 		return CUSTOM_END_MUSIC_BIOMES.contains(biomeName);
 	}
+	
+	/**
+	 * Gets a random ocean biome for a given {@link OceanType} and {@link INoiseRandom}.
+	 *
+	 * @param type An {@link OceanType} to categorize the ocean temperature.
+	 * @param random An {@link INoiseRandom} to randomly pick the ocean variant.
+	 * @return A random ocean biome for a given {@link OceanType} and {@link INoiseRandom}.
+	 */
+	public static RegistryKey<Biome> getOceanBiome(OceanType type, INoiseRandom random) {
+		return OCEAN_BIOME_MAP.getOrDefault(type, new WeightedNoiseList<>()).get(random);
+	}
+	
+	/**
+	 * Get the corresponding deep ocean variant of a {@link Biome} {@link RegistryKey}.
+	 *
+	 * @param oceanBiome The {@link Biome} {@link RegistryKey} to use to get the deep ocean variant.
+	 * @return Null if no deep variant has been registered, otherwise the deep variant.
+	 */
+	@Nullable
+	public static RegistryKey<Biome> getDeepOceanBiome(RegistryKey<Biome> oceanBiome) {
+		return DEEP_OCEAN_BIOME_MAP.get(oceanBiome);
+	}
+	
+	/**
+	 * Check if a {@link Biome} {@link RegistryKey} is an ocean {@link Biome}.
+	 *
+	 * @param biome The {@link Biome} {@link RegistryKey} to check.
+	 * @return If a {@link Biome} {@link RegistryKey} is registered as an ocean {@link Biome}.
+	 */
+	public static boolean isOceanBiome(RegistryKey<Biome> biome) {
+		for (WeightedNoiseList<RegistryKey<Biome>> list : OCEAN_BIOME_MAP.values()) {
+			if (list.getEntries().stream().anyMatch((pair) -> pair.getFirst().equals(biome))) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * Check if a {@link Biome} {@link RegistryKey} is an ocean {@link Biome}, but also not registered as a deep variant.
+	 *
+	 * @param biome The {@link Biome} {@link RegistryKey} to check.
+	 * @return If a {@link Biome} {@link RegistryKey} is registered as an ocean {@link Biome}, but not as a deep variant.
+	 */
+	public static boolean isShallowOceanBiome(RegistryKey<Biome> biome) {
+		for (WeightedNoiseList<RegistryKey<Biome>> list : OCEAN_BIOME_MAP.values()) {
+			if (list.getEntries().stream()
+			.filter((pair) -> DEEP_OCEAN_BIOME_MAP.values().stream().noneMatch((key) -> key.equals(pair.getFirst())))
+			.anyMatch((pair) -> pair.getFirst().equals(biome))) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
I wonder if we should consider optimizing these two methods.
These two methods get called in many areas of Minecraft's layer generation, and any impact to world gen performance should be as little as possible.
One option would be to have two `HashSet`s to store the ocean and shallow ocean `RegistryKey`s to make lookup times much faster.